### PR TITLE
Bugfix for #110

### PR DESCRIPTION
**Pull request by HuanhuanSunMSFT**
The root cause for this issue is due to race condition. The background thread hasn't put the calculated lists into property collection while main thread already proceeds to get it. This fix makes sure this kind of race condition won't crash VS. Also update some other related places.

@EricMSFT @AndreSayreMSFT @adamdriscoll @Microsoft/poshtools

### CODE REVIEW

**Comment by ASayre** [comment]
Will the main thread get a chance to get back to it, or is this just a defensive measure?

**Comment by HuanhuanSunMSFT** [comment]
This is just a defensive measure as all these implementations are actually controlled by VS. When a keystroke occurs, VS will trigger the implementations such as GetClassificationSpans defined in IClassifier. The method needs to get those tokens and corresponding spans. If they are not yet available, then GetProperty will trigger a KeyNotFoundException. 

Ultimate solution to this should be align the sequence of actions between our analysis and VS.

**Comment by ASayre** [comment]
The above looks good, but when/where do we plan to address root cause?

**Comment by HuanhuanSunMSFT** [comment]
With the latest commits, root cause addressed. The original code's issue is it checks if tokenizing is running, if yes, then it just return, which may cause the latest typed char not colorized. 
The fix here is let every keystroke trigger colorization but only apply them when the tracing span is the current one. This ensures that the latest char will always take effect. 
@AndreSayreMSFT

File: PowerShellTools/Classification/PowerShellTokenizationService.cs [hunk 1]
```diff
@@ -268,13 +243,14 @@ public static class BufferProperties
         public const string SpanTokenized = "PSSpanTokenized";
         public const string Regions = "PSRegions";
         public const string RegionTags = "PSRegionTags";
-        public const string Classifier = "PowerShellClassifier";
+        public const string Classifier = "Classifier";
```
Comment (ASayre) [inline PowerShellTools/Classification/PowerShellTokenizationService.cs:246]: Any reason for this change?

**Comment by ASayre** [comment]
Also fixes #228 then? [And potentially, but not necessarily, #126?]

**Comment by HuanhuanSunMSFT** [comment]
Yes, all are fixed in this PR.

**Comment by ASayre** [comment]
Thanks!  For posterity's sake: Our in person convo about 'Classifier' was because one of our classes was named 'classifier', this isn't due to A VS-class named 'classifier'.

**Pull request merged**